lib/fss.ex

```diff
@@ -13,19 +13,65 @@ defmodule FSS do
 
   defmodule S3 do
     defmodule Config do
-      defstruct [:region, :access_key_id, :secret_access_key]
+      defstruct [
+        :access_key_id,
+        :region,
+        :secret_access_key,
+        {:endpoint, "amazonaws.com"},
+        :token
+      ]
 
       @type t :: %__MODULE__{
-              region: String.t(),
               access_key_id: String.t(),
-              secret_access_key: String.t()
+              endpoint: String.t(),
+              region: String.t(),
+              secret_access_key: String.t(),
+              token: String.t() | nil
             }
     end
 
     defmodule Entry do
-      defstruct [:url, :config]
+      defstruct [:bucket, :key, :port, :config]
+
+      @type t :: %__MODULE__{
+              bucket: String.t(),
+              key: String.t(),
+              port: pos_integer(),
+              config: Config.t()
+            }
+
+      def config_from_system_env() do
+        %FSS.S3.Config{
+          access_key_id: System.get_env("AWS_ACCESS_KEY_ID"),
+          secret_access_key: System.get_env("AWS_SECRET_ACCESS_KEY"),
+          region: System.get_env("AWS_REGION", System.get_env("AWS_DEFAULT_REGION")),
+          token: System.get_env("AWS_SESSION_TOKEN")
+        }
```

Comment:
What happens if any of those env vars are missing? Should we try to raise if any of them is missing?

Reply:
We should raise now :boom: